src/App.tsx

```diff
@@ -15,48 +15,112 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+const SORT_FIELD_LIGHT = 'is-light';
+
+enum SortType {
+  Alphabetical = 'alpha',
+  Length = 'length',
+  None = '',
+}
+
+function getPreparedGoods(
+  { sortField, sortedLength, isReversed }: {
+    sortField: SortType,
+    sortedLength: SortType,
+    isReversed: boolean,
+  },
+): string[] {
+  let preparedGoods = [...goodsFromServer];
+
+  if (sortField === SortType.Alphabetical) {
+    preparedGoods = [...preparedGoods].sort(
+      (good1, good2) => good1.localeCompare(good2),
+    );
+  }
+
+  if (sortedLength === SortType.Length) {
+    preparedGoods = [...preparedGoods].sort(
+      (good1, good2) => good1.length - good2.length,
+    );
+  }
+
+  if (isReversed) {
+    preparedGoods = [...preparedGoods].reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState<SortType>(SortType.None);
+  const [sortedLength, setSortedLength] = useState<SortType>(SortType.None);
+  const [isReversed, setIsReversed] = useState<boolean>(false);
+
+  const visibleGoods: string[] = getPreparedGoods(
+    { sortField, sortedLength, isReversed },
+  );
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={sortField === 'alpha' ? 'button is-info'
+            : `button is-info ${SORT_FIELD_LIGHT}`}
+          onClick={() => {
+            setSortField(SortType.Alphabetical);
+            setSortedLength(SortType.None);
+          }}
```

Comment:
consider to make one handler factory for both sort fields in component and just call with proper value in jsx
```
const handleSort = (sortType: SortType) => () => setSortField(sortType);
```
```suggestion
          onClick={handleSort(SortType.Alphabetical)}
```